### PR TITLE
fix create pr, default branch, checkout

### PR DESCRIPTION
**Pull request by phnx47**
- fix create Pull Request
- fix detect default branch in repository
- fix clone repository for correct checkout to custom branch
- push only if we have file changes

### CODE REVIEW

**Comment by phnx47** [comment]
We can try use `--depth 1`, I found how work with fetch with shallow clone: https://stackoverflow.com/a/19528379/5056148

I tried it in my local machine, it works...

File: entrypoint.sh [hunk 1]
```diff
@@ -139,21 +140,24 @@ for repository in "${REPOSITORIES[@]}"; do
...
+        if [ ! -z "$PULL_REQUEST_BRANCH_NAME" -a "$BRANCH_NAME" != "$PULL_REQUEST_BRANCH_NAME" ]; then
+            echo "Creating pull request from [$BRANCH_NAME] into [$PULL_REQUEST_BRANCH_NAME]"
+            jq -n --arg title "File sync from ${GITHUB_REPOSITORY}" --arg head "$BRANCH_NAME" --arg base $PULL_REQUEST_BRANCH_NAME '{title:$title,head:$head,base:$base}' | curl -d @- \
+                -X POST \
+                -H "Accept: application/vnd.github.v3+json" \
+                -u ${USERNAME}:${GITHUB_TOKEN} \
+                --silent \
+                ${GITHUB_API_URL}/repos/${REPO_NAME}/pulls
+        fi
+        cd $TEMP_PATH
```
Comment (kbrashears5) [inline entrypoint.sh]: i think this needs to be outside the if statement, it's used to cleanup folders

File: entrypoint.sh [hunk 2]
```diff
@@ -57,7 +57,7 @@ for repository in "${REPOSITORIES[@]}"; do
     echo "Repository name: [$REPO_NAME]"
 
     echo "Determining default branch name"
-    DEFAULT_BRANCH_NAME=$(curl -X GET -H "Accept: application/vnd.github.v3+json" --silent "${GITHUB_API_URL}/repos/${GITHUB_REPOSITORY}" | jq '.default_branch')
+    DEFAULT_BRANCH_NAME=$(curl -X GET -H "Accept: application/vnd.github.v3+json" -u ${USERNAME}:${GITHUB_TOKEN} --silent "${GITHUB_API_URL}/repos/${REPO_NAME}" | jq -r '.default_branch')
```
Comment (kbrashears5) [inline entrypoint.sh:60]: i didn't think we needed auth for this call - unless this is for supporting private repos?

**Review by kbrashears5 — CHANGES_REQUESTED** [review]
@phnx47 had a few questions, can you update? otherwise looks good

File: entrypoint.sh (hunk 2 above)
Comment (phnx47) [reply entrypoint.sh:60]: Yes, it is for private repository. I tested it in private repository

File: entrypoint.sh (hunk 1 above)
Comment (phnx47) [reply entrypoint.sh]: fixed

**Review by kbrashears5 — APPROVED** [review]

**Pull request merged**